urllib3/_collections.py

```diff
@@ -284,38 +350,46 @@ def _copy_from(self, other):
             self._container[key.lower()] = [key] + val
 
     def copy(self):
+        # type: () -> HTTPHeaderDict
         clone = type(self)()
         clone._copy_from(self)
         return clone
 
     def iteritems(self):
+        # type: () -> Iterator[Tuple[str, str]]
         """Iterate over all header lines, including duplicate ones."""
         for key in self:
             vals = self._container[key.lower()]
             for val in vals[1:]:
-                yield vals[0], val
+                # TODO remove ignore when python/mypy#1578
+                yield vals[0], val  # type: ignore
 
     def itermerged(self):
+        # type: () -> Iterator[Tuple[str, str]]
         """Iterate over all headers, merging duplicate ones together."""
         for key in self:
             val = self._container[key.lower()]
-            yield val[0], ', '.join(val[1:])
+            # TODO remove ignore when python/mypy#1579
+            yield val[0], ', '.join(val[1:])  # type: ignore
 
     def items(self):
+        # type: () -> List[Tuple[str, str]]
         return list(self.iteritems())
 
     @classmethod
     def from_httplib(cls, message):  # Python 2
```

Comment:
As a rule, we prefer duck-typing over version-checking when possible because it's never that simple (there are tons of implementation of Python and they don't all agree).